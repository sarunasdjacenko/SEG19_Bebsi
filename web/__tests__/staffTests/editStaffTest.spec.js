import { shallowMount, RouterLinkStub} from "@vue/test-utils";
import db from "../../src/components/firebaseInit";
// @ts-ignore
import Component from "../../src/components/staff/EditStaff.vue";


describe("Component", () => {
  const wrapper = shallowMount(Component, {
    stubs:{
      RouterLink: RouterLinkStub
    },
    mocks: {
      $route: {
        name: "edit-staff",
        params: { email: "email@example.com" }
      }
    }
  });
  test("is a Vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  }),
  //snapshot test
  test("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  }),
  test("has the right amount of inputs", () => {
    expect(wrapper.findAll("input")).toHaveLength(4);
  }),
  test("inputs have the right value", () => {
    wrapper.setData({
      email: "email",
      name: "name",
      dept: "dept",
      role: "role"
    });
    // The data() sets up properly
    expect(wrapper.vm.email).toMatch("email");
    expect(wrapper.vm.name).toMatch("name");
    expect(wrapper.vm.dept).toMatch("dept");
    expect(wrapper.vm.role).toMatch("role");
  }),
  test("form submits successfully", () => {
    wrapper.setMethods({ updateUser: jest.fn() });
    wrapper.find("#submit").trigger("submit");

    expect(wrapper.vm.updateUser).toHaveBeenCalled();
  }),
  test("all inputs have required attribute", () => {
    var inputArray = wrapper.findAll("input");
    // get if there is an input without the required property
    inputArray = inputArray.filter(input => !input.attributes("required"));
    expect(inputArray).toHaveLength(0);
  });
  db.app.delete()
});
