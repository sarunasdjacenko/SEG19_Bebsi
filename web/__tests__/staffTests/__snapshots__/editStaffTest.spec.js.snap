// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div
  id="edit-employee"
  style="background-color: white; padding: 10px 50px 10px 50px; margin-top: 10px;"
>
  <h3>
    Edit User
  </h3>
   
  <div
    class="row"
  >
    <form
      class="col s12"
    >
      <div
        class="row"
      >
        <div
          class="input-field col s12"
        >
          <span>
            Email:
          </span>
           
          <input
            disabled="disabled"
            required="required"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="row"
      >
        <div
          class="input-field col s12"
        >
          <span>
            Name:
          </span>
           
          <input
            required="required"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="row"
      >
        <div
          class="input-field col s12"
        >
          <span>
            Department:
          </span>
           
          <input
            required="required"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="row"
      >
        <div
          class="input-field col s12"
        >
          <span>
            Role:
          </span>
           
          <input
            disabled="disabled"
            required="required"
            type="text"
          />
        </div>
      </div>
       
      <button
        class="btn"
        id="submit"
        type="submit"
      >
        Submit
      </button>
       
      <a
        class="btn grey"
      >
        Cancel
      </a>
    </form>
  </div>
</div>
`;
